### PR TITLE
add delayed change listener to TextFormatter and TextField

### PR DESCRIPTION
**Pull request by robinschmid**
This could work for any TextFormatter and TextField.

Idea is to add the TextFormatter, use its valueProperty, and force an update after a 2 second delay. This will be at least some time to wait for the user to finish the change and will not trigger to many updates.

### CODE REVIEW

File: javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+   * {@code StringConverter}
+   */
+  protected NumberFormat getNumberFormat() {
+    Locale _locale = locale == null ? Locale.getDefault() : locale;
+
+    if (numberFormat != null) {
+      return numberFormat;
+    } else if (pattern != null) {
+      DecimalFormatSymbols symbols = new DecimalFormatSymbols(_locale);
+      return new DecimalFormat(pattern, symbols);
```
Comment (SteffenHeu) [inline javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java:159]: does it make sense to create a new decimal format all the time?

File: javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+/**
+ * Similar to {@link NumberStringConverter}
+ */
+public class FormatDoubleStringConverter extends StringConverter<Double> {
+
+  final Locale locale;
+  final String pattern;
+  final NumberFormat numberFormat;
+
```
Comment (SteffenHeu) [inline javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java:44]: private

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/DoubleComponent.java [hunk 3]
```diff
@@ -24,43 +24,47 @@
...
     textField = new TextField();
-    textField.setTextFormatter(new TextFormatter<>(new NumberStringConverter(format)));
+    textFormatter = new TextFormatter<>(new FormatDoubleStringConverter(format));
+    FxTextFields.attachDelayedTextFormatter(textField, textFormatter);
+
     textField.setPrefWidth(inputsize);
-    textField.textProperty().bindBidirectional(value, new DoubleStringConverter());
+    textFormatter.setValue(defvalue);
+
+    valueProperty().addListener((o, _, nv) -> logger.info("Double is now " + nv));
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/DoubleComponent.java:60]: remove

File: javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java (hunk 2 above)
Comment (robinschmid) [reply javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java:44]: done and also removed all but numberFormat

File: javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java (hunk 1 above)
Comment (robinschmid) [reply javafx-framework/src/main/java/io/github/mzmine/javafx/components/formatters/FormatDoubleStringConverter.java:159]: pushed into constructor.

File: mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/DoubleComponent.java (hunk 3 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/parameters/parametertypes/DoubleComponent.java:60]: somehow did not push the last commit. Already done

**Pull request merged**